File: Javascript/Linked List/M-2130. Maximum Twin Sum of a Linked List.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function pairSum(head: ListNode | null): number {
  let maxSum = 0;
  const nodeVals: number[] = [];
  while (head) {
    nodeVals.push(head.val);
    head = head.next;
  }
  const n = nodeVals.length;
  for (let i = 0; i < n / 2; i++) {
    maxSum = Math.max(nodeVals[i] + nodeVals[n - 1 - i], maxSum);
  }
  return maxSum;
}

File: Linked List/M-2130. Maximum Twin Sum of a Linked List/M-2130. Maximum Twin Sum of a Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        node_vals = []
        while head:
            node_vals.append(head.val)
            head = head.next

        n = len(node_vals)
        max_pair = 0
        for i in range(n // 2):
            max_pair = max(max_pair, node_vals[i] + node_vals[n - 1 - i])

        return max_pair
